src/interfaces/chain.cpp

```diff
@@ -358,22 +359,38 @@ class ChainImpl : public Chain
     {
         ::uiInterface.ShowProgress(title, progress, resume_possible);
     }
-    std::unique_ptr<Handler> handleNotifications(Notifications& notifications) override
+    std::unique_ptr<Handler> handleNotifications(Notifications& notifications, SnapshotFn snapshot_fn) override
     {
-        return MakeUnique<NotificationsHandlerImpl>(*this, notifications);
+        AssertLockNotHeld(::cs_main);
+        AssertLockNotHeld(::mempool.cs);
```

Comment:
Could use `LOCKS_EXCLUDED` annotations for those?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15719#discussion_r271169011

> Could use LOCKS_EXCLUDED annotations for those?

Added annotations. Effect is kind of limited since this is an override and these mutexes aren't accessible where this is called in the wallet, but it still seems better to have these annotations than not.

Longer term, after more changes like this and #15632, which reduce Chain::Lock usage, wallet code should stop holding these locks altogether.